src/assign.c

```diff
@@ -458,7 +458,7 @@ SEXP assign(SEXP dt, SEXP rows, SEXP cols, SEXP newcolnames, SEXP values)
     coln--;
     SEXP thisvalue = RHS_list_of_columns ? VECTOR_ELT(values, i) : values;
     vlen = length(thisvalue);
-    if (isNull(thisvalue) && !isNull(rows)) error(_("When deleting columns, i should not be provided"));  // #1082, #3089
+    if (isNull(thisvalue) && !isNull(rows)) error(_("Direct assignment of NULL to a list column item is not supported. If you intend to assign NULL to an item in the list column, use list(list(NULL)) in your query."));  // #1082, #3089
```

Comment:
Not sure if this is correct - IIUC, this would error whenever `i` is provided and `RHS` is `NULL`, but the column is not necessarily a list column, you might have to generalize the error message here or add a nested `if` clause for when the specific column is a list column.

Reply:
yeah updated error message to cover both cases 

"Direct assignment of NULL when `i` is provided is not supported. If you intend to assign NULL to a column, use `NULL` without `i`. For list columns, if you intend to assign NULL to a specific item, use `list(list(NULL))` in your query.